### ISSUE TITLE
Search function

### ISSUE THREAD

**Issue by nickbe** [bludit/bludit#82]
A good search function is what I miss most so far.
Without the option to look for a specific entry searchin lots of posts is exhausting.

**Comment by dignajar** [comment]
Hi, yes I know.. It's on the todoList http://forum.bludit.com/viewtopic.php?f=13&t=28

**Comment by nickbe** [comment]
I know ;)
Just trying to push it a little. No offense meant.

Cheers
Nick

**Comment by anaggh** [comment]
Yes I agree. As far as I can remember even Nibbleblog lacked a default search option. A search query which can search through all the posts, posts titles, pages, page titles and tags would be awesome!
Temporarily I am using Google to search through my blog(after its crawled once) with this query

    site:domain.com search_query_here

as a workaround.

**Comment by nickbe** [comment]
Any news on the search function yet?

**Comment by CryssDOTnet** [comment]
I am also looking forwarding to a search function.

**Comment by duke-chai** [comment]
I am also looking forward to a search function.

**Comment by nickbe** [comment]
I'm well aware that searching inside posts is not a trivial task when it comes to performance.
So in a first step, I would consider only searching in post titles, tags and so on.

And I think we really need an internal search function. Bludit is technically so cool - you can carry it around on a usb stick, use old PHP version with only the internal server to run it. Having an external dependency for searching does not feel right to me.

**Comment by nickbe** [comment]
While maintaining our internal news system with bludit, it is now getting increasingly difficult to navigate between hundreds of blog entries without a proper search function.

**Comment by nickbe** [comment]
Anything new regarding an internal search function yet?

**Comment by anaggh** [comment]
Hi guys, I wrote a small search script for personal use. https://github.com/anaggh/Bludit-Search Maybe someone could improve it/convert it to a plugin.

**Comment by nickbe** [comment]
Hey nice. That's definetely a start.
Do you search through all content or only the title and tags?

**Comment by anaggh** [comment]
Searches all content and title. Tags are not searched.
Here's a screenshot:

![capture](https://cloud.githubusercontent.com/assets/9278844/18957481/e8e9be56-867c-11e6-9f2d-8e8dadbf7277.PNG)

**Comment by nickbe** [comment]
I think you should have two search modes. 
Rather than content and title you should either search content+tags+title, or (quicksearch) title+tags.

**Comment by nickbe** [comment]
Hey Diego, can you integrate this into bludit so that the search can be covered by this routine?

**Comment by anaggh** [comment]
Hi, I have rewrote the [search script](https://github.com/anaggh/Bludit-Search) from scratch. Now there is only one search box and button.
* Fixed "Child Page" search results not being displayed.
* Now it searches - titles, tags, username, description, content of both pages and posts.
* It ranks them according to relevance now.

| type | score |
| --- | --- |
| title | +30 points
| tags | +15 points
| description | +10 points
| username | +5 points
| content | +1 point each line

You can add it to your theme after copying search.php by including this snippet in your sidebar.
```
<!-- Search -->
<form role="search" action="<?php echo $Site->url() . 'search.php';?>" target="_blank">
    <input type="text" name="q" placeholder="Search...">
</form>
```

Only thing left now is pagination. If someone can convert this to a plugin or add the pending pagination, please send a pull request.

**Comment by nickbe** [comment]
Fantastic !!! Thanks

**Comment by nickbe** [comment]
Anyone here who is experienced enough to be able to create a real plugin for us based on this?
I think it's time to include this into the bludit code base.

**Comment by Turqueso** [comment]
Is there any news on this subject?

**Comment by Tangol** [comment]
**Anaggh**'s [search](https://github.com/anaggh/Bludit-Search/tree/master/v2) works great on https://github.com/bludit/bludit/commit/8cfd7df062f373dcbc2a272cd60270d9b4b76bd6 .
Can't it be included in the v3.0 core as a plugin?

**Comment by dignajar** [comment]
Implemented in Bludit v3.1.0. Thanks!
